imod/protocols/protocol_etomo.py

```diff
@@ -398,45 +395,47 @@ def createOutput(self):
 
                 newTomogram.setLocation(reconstructTomoFilePath)
                 newTomogram.setTsId(tsId)
-                newTomogram.setSamplingRate(ts.getSamplingRate())
+                newTomogram.setSamplingRate(newPixSize)
 
                 # Set default tomogram origin
-                newTomogram.setOrigin(newOrigin=False)
+                newTomogram.setOrigin(newOrigin=None)
 
                 outputSetOfFullTomograms.append(newTomogram)
+                outputSetOfFullTomograms.update(newTomogram)
                 outputSetOfFullTomograms.write()
                 self._store(outputSetOfFullTomograms)
 
             """Post-processed reconstructed tomogram"""
             posprocessedRecTomoFilePath = self.getFilePath(ts, extension=".rec")
             if os.path.exists(posprocessedRecTomoFilePath):
+                tomoDims, newPixSize = self.getNewPixAndDim(ih, posprocessedRecTomoFilePath)
+                self.debug(f"{posprocessedRecTomoFilePath}: pix = {newPixSize}, dims = {tomoDims}")
                 if outputSetOfPostProcessTomograms is None:
                     outputSetOfPostProcessTomograms = self._createSetOfTomograms()
                     outputSetOfPostProcessTomograms.copyInfo(setOfTiltSeries)
+                    outputSetOfPostProcessTomograms.setSamplingRate(newPixSize)
                     self._defineOutputs(PostProcessTomograms=outputSetOfPostProcessTomograms)
                     self._defineSourceRelation(self.inputSetOfTiltSeries,
                                                outputSetOfPostProcessTomograms)
                 else:
                     outputSetOfPostProcessTomograms.enableAppend()
 
                 newTomogram = tomoObj.Tomogram()
-
-                ih = ImageHandler()
-                outputDim, _, _, _ = ih.getDimensions(posprocessedRecTomoFilePath)
-
                 newTomogram.setLocation(posprocessedRecTomoFilePath)
                 newTomogram.setTsId(tsId)
-                newTomogram.setSamplingRate(self.getPixSizeFromDimensions(outputDim))
+                newTomogram.setSamplingRate(newPixSize)
```

Comment:
I'm not an eTOmo expert,but before we were expecting binning at any eTomo step. There changes assume there is only one possible binning shared with al eTomo steps.... is this correct?

Reply:
no,  tomoDims, newPixSize = self.getNewPixAndDim(ih, posprocessedRecTomoFilePath) is recalculated for each output